src/Shared.CLI/Helpers/PackageDownloader.cs

```diff
@@ -23,9 +24,10 @@ public class PackageDownloader
         /// </summary>
         /// <param name="purl">The package to download.</param>
         /// <param name="httpClientFactory">The <see cref="IHttpClientFactory"/> to use to make the client to use when downloading.</param>
+        /// <param name="provider">The <see cref="IManagerProvider{IManagerMetadata}"/> to use.</param>
         /// <param name="destinationDir">The directory where the package needs to be downloaded to.</param>
         /// <param name="doCaching">Check and use the cache if it exists - create if not.</param>
-        public PackageDownloader(PackageURL purl, IHttpClientFactory? httpClientFactory, string? destinationDir = null, bool doCaching = false)
+        public PackageDownloader(PackageURL purl, IHttpClientFactory? httpClientFactory, IManagerProvider<IManagerMetadata>? provider = null, string? destinationDir = null, bool doCaching = false)
```

Comment:
Do we really need both an IHttpClientFactory and a IManagerProvider? Should the requests not flow through the ManagerProvider (which might itself take an IHttpClientFactory depending on the implementation).

Reply:
That's a good point. I hadn't considered putting the IHttpClientFactory in the provider. I will give that a shot.